query-node/joystream-query-node/schema.graphql

```diff
@@ -1,21 +1,489 @@
+	worker: Worker!
+
+	groups: [CuratorGroup] @derivedFrom(field: "curators")
+}
+
+"A group, that consists of `curators` set"
+type CuratorGroup @entity {
+	"Runtime identifier for group"
+	groupId: BigInt! @unique
+
+	"Activity stats of group"
+	isActive: Boolean!
+
+	"All member curators of this group"
+	curators: [Curator!]
+
+	"All classes this group maintains"
+	maintainerOfClasses: [Class!] @derivedFrom(field: "maintainers")
+}
+
+"A voucher for `Entity` creation"
+type EntityCreationVoucher @entity {
+	"Class to which voucher applies"
+	class: Class!
+
+	"Controller to which this voucher applies"
+	controller: EntityController!
+
+	"How many are allowed in total"
+	maximumEntitiesCount: BigInt!
+
+	"How many have currently been created"
+	entitiesCreated: BigInt!
+}
+
+enum EntityControllerType {
+	LEAD
+	MEMBER
+	MAINTAINERS
+}
+
+"Owner of an `Entity`"
+type EntityController @entity {
+	"Current controller, which is initially set based on who created entity"
+	controllerType: EntityControllerType!
+
+	"Optional field, will be set when EntityControllerType is MEMBER"
+	controllerWhenMember: Member
+
+	"Virtual field for 1:1 relationship"
+	voucher: EntityCreationVoucher @derivedFrom(field: "controller")
+}
+
+"""
+Represents a particular instance of a class,
+hence each entity corresponds to exactly one class
+"""
+type Entity @entity {
+	"Runtime identifier for entity"
+	entityId: BigInt! @unique
+
+	"Current controller, which is initially set based on who created entity"
+	controller: EntityController!
+
+	"Forbid groups to mutate any property value"
+	isFrozen: Boolean!
+
+	"Prevent from being referenced by any entity (including self-references)"
+	isReferenceable: Boolean!
+
+	"The class of this entity"
+	class: Class!
+
+	"""
+	What schemas under which this entity of a class is available, think
+	v.2.0 Person schema for John, v3.0 Person schema for John
+	"""
+	supportedSchemas: [Schema!]
+
+	"Values for properties on class that are used by some schema used by this entity!"
+	values: [PropertyValue!]
+
+	"Total number of inbound references from another entities"
+	totalNumberOfInboundReferences: BigInt!
+
+	"Number of inbound references from another entities with 'SameOwner' flag set"
+	sameOwnerInboundReferences: BigInt!
+}
+
+"Fields encode a flattened representation of non-vector properties, only one will be non-null"
+type SingleValue @variant {
+	bool: Boolean
+
+	integer: BigInt
+
+	text: Bytes
+
+	hash: Bytes
+
+	reference: Entity
+}
+
+"""
+Fields encode a flattened representation of vector properties, only one will be non-null
+"""
+type VectorValue @variant {
+	boolVec: [Boolean!]
+
+	intVec: [BigInt!]
+
+	textVec: [Bytes!]
+
+	hashVec: [Bytes!]
+
+	referenceVec: [Entity!]
+
+	"Used to avoid vector data race update conditions"
+	nonce: BigInt!
+}
+
+union StoredValue = SingleValue | VectorValue
+
+type PropertyValue @entity {
+	"Runtime identifier of property value in entity"
+	propertyValueId: BigInt! @unique
```

Comment:
There is no unique property value id from the chain, the id in the chain is just an index in the given class, which will not be unique.

A property value is unique for a given combination and entity&property _of the same class_.

Reply:
Done in d8b9033. Since `PropertyValue` has a reference to `Property` no longer needed that field.

Reply:
There needs to be a `propertyValueId`, because it tells you the index of the value for the given entity. However, This value is no _unique_ across all `PropertyValue` instances, precisely because lots of entities will have a property with index 0,1,2,...

If this is not clear, please review content director data model again.